loops-arrays-objects.js

```diff
@@ -47,66 +39,118 @@ for (const score of scores) {
         console.log("B");
     } else if (score.score < 100) {
         console.log("A");
-    } else {
-        console.log(score.score)
     }
 }
 
+// Calculate and print the average score
+const averageScore = totalScore / scores.length;
+console.log("Average Score:", averageScore);
 
 
 // ==========================================
 // Opdracht 1c
-// Breid je script uit door de bijbehorende letter op te slaan in de 'grade'-property van ieder student-object in de array.
-// Vóór het script zie je de originele objecten,
-// Na jouw script zie je de aangepaste objecten:
-// console.log(scores) geeft: [
-//  { name: 'Max', score: 83, grade: 'B' },
-//  { name: 'David', score: 77, grade: 'C' },
-//  { name: 'Khalid', score: 92, grade: 'A' },
-//  { name: 'Rianne', score: 66, grade: 'D' }
-//  ];
-// ==========================================
+const scores = [
+    {name: 'Max', score: 83, grade: null},
+    {name: 'David', score: 77, grade: null},
+    {name: 'Khalid', score: 92, grade: null},
+    {name: 'Rianne', score: 66, grade: null}
+];
 
-for (let i = 0; i < scores.length; i++) {
-    let score = scores[i];
-    if (score.score < 60) {
-        scores[i] = {name: score.name, score: score.score, grade: "F"};
-    } else if (score.score < 70) {
-        scores[i] = {name: score.name, score: score.score, grade: "D"};
-    } else if (score.score < 80) {
-        scores[i] = {name: score.name, score: score.score, grade: "C"};
-    } else if (score.score < 90) {
-        scores[i] = {name: score.name, score: score.score, grade: "B"};
+// Update the grade property for each student
+function updateGrades() {
+    for (let i = 0; i < scores.length; i++) {
+        let score = scores[i];
+        if (score.score < 60) {
+            score.grade = "F";
+        } else if (score.score < 70) {
+            score.grade = "D";
+        } else if (score.score < 80) {
+            score.grade = "C";
+        } else if (score.score < 90) {
+            score.grade = "B";
+        } else {
+            score.grade = "A";
+        }
+    }
+}
+
+// Initial grade update
+updateGrades();
+
```

Comment:
Na de aanroep van de functie updateGrades(); volstaat console.log(scores);
 Alles daarna is overbodig en displayAllstudents(); is niet gedefinieerd.